### PR TITLE
Allow override for requesting specific worker types for Complement on command line.

### PR DESCRIPTION
**Pull request by realtyem**
## Why
Complement is great. It's really useful to test functionality. You can test monolith and workerised setups. But maybe the test I want to run doesn't need a full set of all types of workers, only the worker that targets a specific function or Servlet. Let's allow for overriding this at the command line level.

This means that if I wish to test that `federation_sender` functionality is correct with a worker, I wouldn't have to start up a bunch of extra workers. I could just run(for example):
```sh
POSTGRES=1 WORKERS=1 WORKER_TYPE="federation_sender" COMPLEMENT_DIR=../complement ./scripts-dev/complement.sh -run TestOutboundFederationSend
```
## Checklist
* [x] Add override functionality to `start_for_complement.sh`
* [x] Update docs to show how this works

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
  - Feel free to credit yourself, by adding a sentence "Contributed by @github_username." or "Contributed by [Your Name]." to the end of the entry.
* [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))
Signed-off-by: Jason Little <realtyem@gmail.com>

### CODE REVIEW

File: docs/development/contributing_guide.md [hunk 1]
```diff
@@ -324,7 +324,7 @@ The above will run a monolithic (single-process) Synapse with SQLite as the data
 
 - Passing `POSTGRES=1` as an environment variable to use the Postgres database instead.
 - Passing `WORKERS=1` as an environment variable to use a workerised setup instead. This option implies the use of Postgres.
-
+  - If setting `WORKERS=1`, optionally set `WORKER_TYPES=` to declare which worker types you wish to test. A simple comma-delimited string containing the worker types defined from the template in [here](https://github.com/matrix-org/synapse/blob/develop/docker/configure_workers_and_start.py). A safe example would be `WORKER_TYPES="federation_inbound,federation_sender,synchrotron"`. See the [worker documentation](https://matrix-org.github.io/synapse/latest/workers.html) for additional information on workers.
```
Comment (dklimpel) [inline docs/development/contributing_guide.md:327]:
```suggestion
  - If setting `WORKERS=1`, optionally set `WORKER_TYPES=` to declare which worker
    types you wish to test. A simple comma-delimited string containing the worker types
    defined from the template in
    [here](https://github.com/matrix-org/synapse/blob/develop/docker/configure_workers_and_start.py).
    A safe example would be `WORKER_TYPES="federation_inbound,federation_sender,synchrotron"`.
    See the [worker documentation](../workers.md) for additional information on workers.
```

**Review by reivilibre — APPROVED** [review]

**Review by reivilibre — CHANGES_REQUESTED** [review]
Not really your fault, but do you think you could make `configure_workers_and_start.py` give an error if you try to create a worker type which isn't a real worker?
I specified `WORKER_TYPES=nonsense` and nothing stopped me. I worry this will lead someone to have a bad day if they make a typo.

**Comment by realtyem** [comment]
It should already [log](https://github.com/matrix-org/synapse/blob/develop/docker/configure_workers_and_start.py#L424) a message that pops up from `configure_workers_and_start.py`. You think a hard stop would be better? I'll put it into #14294

**Comment by reivilibre** [comment]
> It should already [log](https://github.com/matrix-org/synapse/blob/develop/docker/configure_workers_and_start.py?rgh-link-date=2022-11-04T19%3A35%3A02Z#L424) a message that pops up from `configure_workers_and_start.py`. You think a hard stop would be better? I'll put it into #14294

Yeah, I think a hard stop would be better because a mere log line is very easy to miss (especially in Complement output...) and arguably it renders the test null and void / it's not sensible to continue to test workers types that aren't understood.

**Review by reivilibre — APPROVED** [review]

**Pull request merged**